src/driver/amdxdna/amdxdna_drv.c

```diff
@@ -294,6 +306,10 @@ static int amdxdna_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		goto failed_dev_fini;
 	}
 
+	pm_runtime_set_autosuspend_delay(dev, AMDXDNA_AUTOSUSPEND_DELAY);
+	pm_runtime_use_autosuspend(dev);
+	pm_runtime_allow(dev);
```

Comment:
If drm_dev_register() failed, should driver rewind these operations or not?

Reply:
call pm_runtime_forbid() on error.